src/Components/Server/src/ComponentHub.cs

```diff
@@ -7,15 +7,35 @@
 using Microsoft.AspNetCore.Components.Server.Circuits;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Routing;
 using Microsoft.AspNetCore.SignalR;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.AspNetCore.Components.Server
 {
-    /// <summary>
-    /// A SignalR hub that accepts connections to an ASP.NET Core Components application.
-    /// </summary>
+    // Some notes about our expectations for error handling:
+    //
+    // In general, we need to prevent any client from interacting with a circuit that's in an unpredictable
+    // state. This means that when a circuit throws an unhandled exception our top priority is to
+    // unregister and dispose the circuit. This will prevent any new dispatches from the client
+    // from making it into application code.
+    //
+    // As part of this process, we also notify the client (if there is one) of the error, and we
+    // *expect* a well-behaved client to disconnect. A malicious client can't be expected to disconnect,
+    // but since we've unregistered the circuit they won't be able to access it anyway. When a call
+    // comes into any hub method and the circuit has been disassociated, we will abort the connection.
+    // It's safe to assume that's the result of a race condition or misbehaving client.
```

Comment:
So I guess a misbehaving client can, at most, go silent to avoid disconnection, and then can receive at most N more render batches (where N = the limit @javiercn recently implemented, since to receive more it would have to send an ACK).

But the circuit is still *running* during that period, and even beyond it since the client can stay connected indefinitely. Might it be worth having the server proactively hang up, say, 1 second after dispatching the error notification? We just have to give enough time for the message to be definitely dispatched to the network, even if it hasn't yet been delivered to the client.

Reply:
Oh wait, maybe I was misunderstanding. In a later commit, it looks like the circuit is in fact terminated when the fatal error occurs (so no more code inside the components will run), and it's only the SignalR connection that is left open until the next incoming message. Is that correct?

Reply:
Correct.

> Might it be worth having the server proactively hang up, say, 1 second after dispatching the error notification?

The server cannot hang up until it receives a hub message. Put another way, we only have the ability to abort the connection while processing a hub message.

So in the scenario you're pointing out - client causes an error and then goes silent:
1. Error happens
2. Client is notified
3. Circuit is disposed
4. Client goes silent
5. Client can keep the connection open indefinitely, but they don't hold any resources.

In the end this because the same as opening a connection and never calling `StartCircuit`.